infrastructure/storage-provisioning/README.md

```diff
@@ -85,3 +85,31 @@ After that you can delete test with commands
 $ kubectl delete -f examples/wordpress.yaml
 $ kubectl delete -f examples/mysql.yaml
 ```
+
+## Rook NFS
+The personal storage disk attached to VMs and containers is bassd on the NFS server provided by Rook-Ceph ([here](https://rook.io/docs/rook/v1.10/Storage-Configuration/NFS/nfs/)), which can be configured with the `CephNFS` CRD of the NFS-Ganesha server.
+
+The user's storage is dynamically provisioned through a `StorageClass` that contains a set of parameters that tell the Rook Operator to create also a NFS share; this is later mounted from user's VMs and containers though the NFS protocol.
+
+To create the necessary resources:
+```bash
+$ kubectl create -f nfs-manifests/nfs.yaml
+$ kubectl create -f nfs-manifests/storageclass.yaml
+```
+
+Be aware that to use this feature the system must be configured with:
+- Rook v1.9.1 or greather
+- Ceph v16.2.7 or greather
+
+### Using the NFS service
+Using the NFS service inside VirtualMachines requires the `nfs-common` package for Debian based linux OSs and the `nfs-utils` for CentOS based VMs.
+
+The DNS resolution for the service name on Container instances is done on the host machine, which does not have access to the Kubernetes cluster's DNS server.
+To allow the host machine to resolve the DNS name for the NFS server you need to add the proper record to the host's `/etc/resolv.conf`, either manually or with an automated process (e.g., cloud-init).
+
+An example view of `/etc/resolv.conf` with the proper record to resolve the DNS name for the NFS server:
+```bash
+$ cat /etc/resolv.conf
+
+rook-ceph-nfs-rook-nfs-primary-rook-ceph.cluster.local 8.8.8.8
```

Comment:
VM's are able to resolve the DNS name of the NFS service because they do it from inside the cluster itself.
On the other hand for the Containers the resolution is done by the machine that hosts the Kubelet, I don't know why it is done this way, but I've seen from tests that this is the way it works (also with Iorio we've seen it).
So a workaround is to add to the machines that host the Kubelet the DNS resolution for the NFS service with the cluster IP or the DNS resolution for the Kubernetes internal DNS server (to resolve all kind of services inside the cluster).
It is not done automatically. We've talked about a possible solution using an Ansible dedicated to this purpose, but has not been implemented by us.

Reply:
Got it, I slightly modified the text. Please check that is ok.
Still open the other question: "Which IP address do I have to add? The cluster-IP one?"

Reply:
Yes the cluster IP